ProjectLibertyLabs/gateway
# add ability to get account given public key

# Problem
Currently, we cannot get an msa or any other account information given a public key. We need to add functionality to do so.

Link to GitHub Issue(s): #266 

# Solution
- Added endpoint and service to support getting an account given a public key
- Added a service to get an msaId given a public key

## Steps to Verify:

1. Spin up swagger
2. Test that endpoint return correct account.
3. See that open api displays all of the correct info.

## Screenshots (optional):
<img width="1492" alt="Screenshot 2024-08-01 at 3 28 19 PM" src="https://github.com/user-attachments/assets/b94b79a8-d8f6-4040-acd3-ae4f2292bb5d">

## Additional Info
While the ticket states to just return an MSA Id, I thought it would be more complete to return the associated account, which is an object that contains the msaId and handle.
